chia/util/keyring_wrapper.py

```diff
@@ -195,13 +259,52 @@ def set_master_passphrase(
                 self.keyring.load_keyring(passphrase=current_passphrase)
                 self.keyring.write_keyring(fresh_salt=True)  # Create a new salt since we're changing the passphrase
 
+        if supports_os_passphrase_storage():
+            if save_passphrase:
+                self.save_master_passphrase_to_credential_store(new_passphrase)
+            else:
+                self.remove_master_passphrase_from_credential_store()
+
     def remove_master_passphrase(self, current_passphrase: Optional[str]) -> None:
         """
         Remove the user-specific master passphrase. We still keep the keyring contents encrypted
         using the default passphrase.
         """
         self.set_master_passphrase(current_passphrase, DEFAULT_PASSPHRASE_IF_NO_MASTER_PASSPHRASE)
 
+    def save_master_passphrase_to_credential_store(self, passphrase: str) -> None:
+        if platform == "darwin":
+            mac_keychain = MacKeyring()
+            try:
+                mac_keychain.set_password(
+                    MAC_KEYCHAIN_MASTER_PASSPHRASE_SERVICE, MAC_KEYCHAIN_MASTER_PASSPHRASE_USER, passphrase
+                )
+            except KeyringError as e:
+                warn_if_macos_errSecInteractionNotAllowed(e)
+        return None
+
+    def remove_master_passphrase_from_credential_store(self) -> None:
+        if platform == "darwin":
+            mac_keychain = MacKeyring()
+            try:
+                mac_keychain.delete_password(
+                    MAC_KEYCHAIN_MASTER_PASSPHRASE_SERVICE, MAC_KEYCHAIN_MASTER_PASSPHRASE_USER
+                )
+            except KeyringError as e:
+                warn_if_macos_errSecInteractionNotAllowed(e)
+        return None
+
+    def get_master_passphrase_from_credential_store(self) -> Optional[str]:
```

Comment:
Access to the master keys is only available to other processes on the same machine, or to any connecting websocket process with the correct SSL key?

Reply:
It's a bit nuanced. Private keys (in keyring.yaml) are accessible to processes that connect to the daemon using the correct SSL key, provided that the daemon has access to the master passphrase. The daemon gains access to the master passphrase either through user interaction (on the command line or via the GUI), or by fetching the saved passphrase from the macOS Keychain or Windows Credential Manager (saving the passphrase is optional).

Previously, private keys were freely readable by any Python process on macOS or Windows, and any process capable of reading ~/.local/share/python-keyring/cryptfile_pass.cfg on Linux.